From 96e72f4bbb380bdc543770d12345989367899577 Mon Sep 17 00:00:00 2001
From: Raja Ganapathi Busam <quic_rbusam@quicinc.com>
Date: Wed, 5 Mar 2025 15:32:42 +0530
Subject: [PATCH 4/4] waylandsink: increase timeout limitation in
 gst_wl_window_new_toplevel

- Issue: There is no enough time for xdg surface configuring in stress
         tests of activate-deactivate example and EIS case. In this
         case, it will report "xdg_toplevel@17: error 1: Surface has not
         been configured yet".
- Fix: Increase timeout limitation in gst_wl_window_new_toplevel.

Upstream-Status: Inappropriate [Qualcomm specific]

Signed-off-by: Raja Ganapathi Busam <quic_rbusam@quicinc.com>
---
 gst-libs/gst/wayland/gstwlwindow.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst-libs/gst/wayland/gstwlwindow.c b/gst-libs/gst/wayland/gstwlwindow.c
index 27084c0..c294064 100644
--- a/gst-libs/gst/wayland/gstwlwindow.c
+++ b/gst-libs/gst/wayland/gstwlwindow.c
@@ -344,7 +344,7 @@ gst_wl_window_new_toplevel (GstWlDisplay * display, const GstVideoInfo * info,
     wl_display_flush (gst_wl_display_get_display (display));
 
     g_mutex_lock (&priv->configure_mutex);
-    timeout = g_get_monotonic_time () + 100 * G_TIME_SPAN_MILLISECOND;
+    timeout = g_get_monotonic_time () + 200 * G_TIME_SPAN_MILLISECOND;
     while (!priv->configured) {
       if (!g_cond_wait_until (&priv->configure_cond, &priv->configure_mutex,
               timeout)) {
-- 
2.25.1

